### ISSUE TITLE
[Serializer] Serialization of empty ArrayCollection to JSON returns '{}' instead '[]'

### ISSUE THREAD

**Issue by yauhenko** [symfony/symfony#37061]
**Symfony version(s) affected**: 5.0.9, 5.1.0

**Description**  
Entity User
```
class User implements UserInterface {

	/**
	 * @ORM\Id()
	 * @ORM\GeneratedValue()
	 * @ORM\Column(type="integer", options={"unsigned": true})
	 * @Groups("main")
	 */
	private int $id;
       // ...
	/**
	 * @ORM\ManyToMany(targetEntity=Department::class)
	 * @ORM\JoinColumn(onDelete="CASCADE")
	 * @Groups("main")
	 */
	private $departments;
```
My controller
```
	public function getUser(User $id) {
            return $this->json($user, 200, [], ['groups' => ['main']);
	}
```
Expected:
```
{
    "id": 2,
   // ...
    "departments": []
  }
```

Actual result:
```
{
    "id": 2,
   // ...
    "departments": {}
  }
```

**How to reproduce**  
Just upgraded symfony/serializer to 5.0.9

**Comment by yceruto** [comment]
Duplicated of https://github.com/symfony/symfony/issues/37041? and already fixed in https://github.com/symfony/symfony/pull/37049

**Comment by PlaginDL** [comment]
This thing is not yet fixed in 5.0 and 5.1 branches.

**Comment by guillaumesmo** [comment]
> This thing is not yet fixed in 5.0 and 5.1 branches.

it will be in the next patch releases (4.4.10, 5.0.10 and 5.1.1) usually at the end of the month

**Comment by arisrais** [comment]
Thanks, I am waiting for it. I will bypass serializer v4.4.9 (my CI is down)
